trieste/acquisition/function.py

```diff
@@ -600,13 +600,14 @@ def prepare_acquisition_function(
 
         constraint_fn = self._constraint_builder.prepare_acquisition_function(datasets, models)
         pof = constraint_fn(objective_dataset.query_points[:, None, ...])
-        is_feasible = pof >= self._min_feasibility_probability
+        is_feasible = tf.squeeze(pof >= self._min_feasibility_probability, axis=-1)
```

Comment:
i'll be honest - i'm a bit vague on this. It makes sense to me and it passes the tests, but I'm not in the place to think it through carefully atm